### PR TITLE
remove unwrap from tokio server

### PR DESCRIPTION
**Pull request by AnderEnder**

### CODE REVIEW

File: diamond/src/bin/diamond-server.rs [hunk 1]
```diff
@@ -50,14 +50,19 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let tcp_server = async move {
         let mut incoming = tcp_listener.incoming();
         while let Some(sock) = incoming.next().await {
-            let mut framed_sock = Framed::new(sock.unwrap(), LinesCodec::new());
-            let conf = config_tcp.clone();
+            match sock {
+                Ok(sock) => {
+                    let mut framed_sock = Framed::new(sock, LinesCodec::new());
+                    let conf = config_tcp.clone();
```
Comment (andy128k) [inline diamond/src/bin/diamond-server.rs]: This is not needed anymore. just use `config_tcp`.

File: diamond/src/bin/diamond-server.rs [hunk 2]
```diff
@@ -50,14 +50,19 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
 
-            while let Some(line) = framed_sock.next().await {
-                match line {
-                    Ok(line) => update_silently(&line, &conf),
-                    Err(e) => eprintln!("accept error = {:?}", e),
+                    while let Some(line) = framed_sock.next().await {
+                        match line {
+                            Ok(line) => update_silently(&line, &conf),
+                            Err(e) => eprintln!("accept error = {:?}", e),
```
Comment (andy128k) [inline diamond/src/bin/diamond-server.rs]: This is not an 'accept error', but 'recv error' or smth.

File: diamond/src/bin/diamond-server.rs (hunk 1 above)
Comment (AnderEnder) [reply diamond/src/bin/diamond-server.rs]: done

File: diamond/src/bin/diamond-server.rs (hunk 2 above)
Comment (AnderEnder) [reply diamond/src/bin/diamond-server.rs]: done

**Pull request merged**